frontend/src/Navigation.ts

```diff
@@ -85,3 +94,17 @@ export default async ({ dynamicNav, currentNamespace }: DynamicNav) => {
       }
     : routes;
 };
+
+export const useNavigation = ({ dynamicNav, currentNamespace }: DynamicNav): Navigation | RouteProps[] => {
+  const [navigation, setNavigation] = React.useState<Navigation | RouteProps[]>();
+  React.useEffect(() => {
+    if (dynamicNav) {
+      // this is just one off for now, but we can start building on this
+      calculateNavigation({ dynamicNav, currentNamespace }).then((data: Navigation | RouteProps[]) => setNavigation(data));
+    }
+  }, [dynamicNav]);
```

Comment:
Hmmm, you're missing `currentNamespace` from your dependencies. Any reason why? I'm surprised we don't have a lint to prevent this 🤔

Reply:
@karelhala I think we should add this to the hook. It's data from outside our app and we are using it blindly as part of the URL. It probably won't change in the lifecycle of this usage, but we should write it as such that we do not know what or when it will change as we have no control over the other side of the fence from this project.

Reply:
Yeah! Good point.